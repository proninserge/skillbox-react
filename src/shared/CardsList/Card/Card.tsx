import { CardHeader } from './CardHeader/CardHeader';
import { CardControls } from './CardControls/CardControls';
import { CardMenu } from './CardMenu/CardMenu';
import styles from './card.less';

const postId = '123';

export function Card() {
    return (
        <li className={styles.card}>
            <CardHeader />
            <div className={styles.postPreview}>
                <img src="https://cdn.dribbble.com/users/730703/screenshots/15503300/media/384ddbf6e843acf7457b5946f053df76.jpg?compress=1&resize=300x225" alt="Post preview"/>
            </div>
            <CardMenu postId={postId} />
            <CardControls />
        </li>
    );
}