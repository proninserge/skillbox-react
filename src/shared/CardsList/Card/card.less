@import '~@/variables.global.less';

.card {
    box-sizing: border-box;
    background-color: @white;
    border-radius: 7px;
    display: flex;
    flex-flow: row wrap;
    position: relative;
    margin-bottom: 26px;

    &:last-of-type {
        margin: 0;
    }

    @media all and (min-width: 1024px) {
        padding: 0 40px;
        align-items: center;
    }
}

.card > * {
    flex: 1 1 100%;
}

.postPreview {
    max-height: 178px;
    overflow: hidden;

    img {
        display: block;

        width: 100%;
        height: 100%;
        object-fit: cover;
    }

    @media all and (min-width: 1024px) {
        order: 1;
        max-width: 190px;

        img {
            width: 100%;
            height: auto;
        }
    }
}
