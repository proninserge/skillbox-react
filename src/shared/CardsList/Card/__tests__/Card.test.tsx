import {shallow, configure} from 'enzyme';
import {Card} from '../Card';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Card', () => {
    const wrapper = shallow(<Card author={'random'} title={'random'} id={'random'} preview={{images: [{source: {url: 'random', width: 'random', height: 'random'}}]}} created_utc={'random'} thumbnail={'random'} ups={'random'} downs={'random'} />);

    test('should render with a preview box', () => {
        expect(wrapper).toBeDefined();
        expect(wrapper.find('div.postPreview').isEmptyRender()).toBeFalsy();
    });

    test ('should render snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

});