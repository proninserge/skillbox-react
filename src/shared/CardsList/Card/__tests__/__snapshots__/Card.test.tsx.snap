// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render snapshot 1`] = `
<li
  className="card"
>
  <CardHeader />
  <div
    className="postPreview"
  >
    <img
      alt="Post preview"
      src="https://cdn.dribbble.com/users/730703/screenshots/15503300/media/384ddbf6e843acf7457b5946f053df76.jpg?compress=1&resize=300x225"
    />
  </div>
  <CardMenu
    postId="123"
  />
  <CardControls />
</li>
`;
